lib/ruby_lsp/rubocop/runtime_adapter.rb

```diff
@@ -42,6 +52,14 @@ def uri_to_path(uri)
           uri.to_s.delete_prefix('file://')
         end
       end
+
+      def prism_result(document)
+        prism_result = document.parse_result
```

Comment:
@vinistock I have a question. Currently, the return value of `document.parse_result` is an instance of `Prism::ParseResult`.  Is there a plan to change this return value to an instance of `Prism::ParseLexResult`?  

Alternatively, is there a plan to add an API like `document.parse_lex_result` to obtain an instance of `Prism::ParseLexResult`?  

In any case, since `Prism::Translation::Parser#tokenize` requires an instance of `Prism::ParseLexResult`, not `Prism::ParseResult`, I expect that some form of adjustment will be made.

Reply:
You're correct. To make this work, we are going to need to change our document implementation to always use the `parse_lex_result`, so that documents have the information necessary.

I'll look into making the changes in the LSP.